src/pvdr/kprobe.c

```diff
@@ -271,28 +282,28 @@ static int kprobe_attach_pattern(kprobe_t *kp, const char *pattern)
 	return (err < 0) ? err : kp->efds.len;
 }
 
-static int kprobe_load(node_t *probe, prog_t *prog, const char *type)
+static int kprobe_load(node_t *probe, prog_t *prog, const char *probestring,
+		       const char *type, kprobe_t **kpp)
 {
-	kprobe_t *kp;
 	char *func;
 
-	kp = probe_load(BPF_PROG_TYPE_KPROBE, probe, prog);
-	if (!kp)
+	*kpp = probe_load(BPF_PROG_TYPE_KPROBE, probe, prog);
```

Comment:
i think it's cleaner to keep 'kp' variable and assign *kpp once instead of dereferencing it everywhere